src/Docfx.MarkdigEngine.Extensions/MarkdownContext.cs

```diff
@@ -51,6 +57,11 @@ public class MarkdownContext
     /// </summary>
     public GetImageLinkDelegate GetImageLink { get; }
 
+    /// <summary>
+    /// Get the configuration for a given extension
+    /// </summary>
+    public GetExtensionConfigurationDelegate GetExtensionConfiguration { get; }
```

Comment:
Consider keep the existing pattern of passing plant UML options to `UsePlantUml` and `UseDocfxExtensions` as additional arguments. `IReadOnlyDictionary<string, string>` isn't a good abstraction for extension configurations - we may in the future have extension configs that are more than string -> string maps.

Reply:
For me the extension owns/understands its configuration.  The consumer of the extension should not be burdened by knowing how to configure the extension.  This is internal knowledge to the extension.  The extension will pull the relevant configuration using the delegate that provides a map (key/value pair) of configuration parameters from the `MarkdownContext`.  The consumer just passes the map on and the extension takes care of the rest

Reply:
I mean `IReadOnlyDictionary<string, string>` cannot represent the config if it uses `bool`, `number`, `array` or nested objects such as:

```json
{
  "my-extension": {
    "showTable": false,
    "tableOptions": {
      "displayAs": "table"
    }
  }
}
```

Reply:
Are you proposing `Dictionary<string, object>` rather?

Reply:
I've changed the delegate signature to return an `object` 

```c#
    /// <summary>
    /// Allows configuration of extensions
    /// </summary>
    /// <param name="extension">Name of the extension being configured</param>
    public delegate object GetExtensionConfigurationDelegate(string extension);
```

The extension can then pull it using code such as:

```c#
    public PlantUmlExtension(MarkdownContext context)
    {
        _context = context;
        _settings = new();

        if (_context.GetExtensionConfiguration("PlantUml") is Dictionary<string, string> config)
            _settings = new DocfxPlantUmlSettings(config);
    }
```

This will enable any .NET intrinsic type to be used.